### PR TITLE
Live Plugins Pilot [DOC-791]

### PR DESCRIPTION
**Pull request by pwseg**
### Proposed changes

- New doc for Live Plugins pilot.

### Merge timing

- 1 February 2024.

### CODE REVIEW

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+---
+
+Live plugins are JavaScript code snippets published to your Segment workspace and then downloaded directly to the mobile devices of end users. Live plugins let you perform real-time modifications to events before they leave the mobile device.
+
+On this page, you'll learn how to set up live plugins and how to create your own live plugins. You'll also see example live plugins that address common use cases.
+
+> info "Live Plugins is in pilot"
+> Live Plugins is currently in Pilot and available to select Business Tier Customers only. To enable this feature for your workspace, contact your CSM.
+
+## Live Plugins overview
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:13]:
```suggestion
## Live plugins overview
```

Comment (stayseesong) [reply src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:13]: Can we have it just say "Overview"?

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+
+You can use JavaScript live plugins with Analytics-Swift and Analytics-Kotlin to filter and modify data remotely. As a result, you can filter and modify analytics events without having to deploy updates to the app store for each change, ensuring data quality and consistency for all your mobile users.
+
+Because live plugins let you modify event data before it leaves a mobile device, you can use the same function to modify data meant for all your cloud-mode and device-mode destinations. 
+
+## Live plugin setup
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:19]: can we just have it say "Setup"?

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+To use live plugins, you first need to set up your mobile app with a one-time configuration.
+
+### One-time configuration
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:23]: Is this heading necessary? Can all the steps listed under just be under the setup heading?

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+
+Follow these steps to configure Live Plugins:
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:25]:
```suggestion
To configure live plugins:
```

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+// Instantiate analytics
+// Add LivePluginsto Analytics
+analytics.add(LivePlugins())
+```
+{% endcodeexampletab %}
+{% endcodeexample %}
+
+After you've completed setup, you can deploy your apps to the Apple App Store and Google Play Store. You can then add new JavaScript plugin code to your mobile apps through the CLI and perform updates as often as needed.
+
+## Live plugin tutorial
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:56]: the sub-headings in this section are a little confusing, as it looks like they're steps, but the headings don't really make that clear that they're steps that need to be completed in order

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+
+#### 1. Install the CLI with Homebrew
+
+Run this command to install the Segment CLI:
+
+```shell
+$ brew install segment-integrations/formulae/segmentcli
+```
+
+#### 2. Authenticate witih Segment
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:90]:
```suggestion
#### 2. Authenticate with Segment
```

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+You've now successfully attached your live plugin(s) to your mobile source. The next time your users launch your app, their Segment SDK will download the latest live plugins, which will run every time new events are generated.
+
+> info ""
+> Because the CDN settings object takes a few minutes to rebuild, your live plugins might not be available immediately.
+
+## Create your own live plugin
+
+Follow the steps in this section to create your own live plugin.
+
+### 1. Subclass `LivePlugin`
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:117]:
```suggestion
### 1. Subclass the `LivePlugin` class
```

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+
+When you adding a new instance, you specify the `LivePluginType` and the destination to which it applies, or use null to apply it to all destinations.
+
+Here's how you can add the `UserIdLivePlugin` to your Analytics instance:
+
+```js
+analytics.add(new UserIdLivePlugin(LivePluginType.enrichment, "adobe"));
+```
+
+### 3. Using `LivePluginType` enums
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:154]:
```suggestion
### 3. Use the `LivePluginType` enums
```

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+analytics.add(new DownSampleLivePlugin(LivePluginType.enrichment, null));
+
+```
+
+{% endcodeexampletab %}
+{% endcodeexample %}
+
+## Live Plugins API
+
+Live plugins follow an interface that lets you intercept Segment events and modify their data. This interface includes several functions that you can implement for custom behavior:
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:306]:
```suggestion
Live plugins follow an interface that let you intercept Segment events and modify their data. This interface includes several functions that you can implement for custom behavior:
```

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+| `execute(event): event` | Called for every event. Must return the event or `null` to drop it. If you do return the event, then the more specific callback based on the event type is called. Use this callback if the event type isn't important. Additionally, you can call `super.execute()` to use one of the event type callbacks for Track, Identify, Screen, Group, or Alias calls.  |
+
+#### Additional event callbacks
+
+The following callback functions are designed for specific event types and let you control event modification:
+
+| Callback          | Description                                                                                   |
+| ----------------- | --------------------------------------------------------------------------------------------- |
+| `track(event)`    | Called for a tracking event. Must return the event to keep it or return `null` to drop it.    |
+| `identify(event)` | Called for an identified event. Must return the event to keep it or return `null` to drop it. |
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:345]:
```suggestion
| `identify(event)` | Called for an identify event. Must return the event to keep it or return `null` to drop it. |
```

**Review by stayseesong — APPROVED** [review]
@pwseg left some comments

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md (hunk 2 above)
Comment (pwseg) [reply src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:19]: Yep, done.

File: src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md (hunk 3 above)
Comment (pwseg) [reply src/connections/sources/catalog/libraries/mobile/apple/live-plugins.md:23]: Agreed, it definitely reads better without this heading.

**Pull request merged**